src/qt/bitcoin.cpp

```diff
@@ -441,6 +440,7 @@ void BitcoinApplication::initializeResult(bool success, interfaces::BlockAndHead
 #endif
         pollShutdownTimer->start(SHUTDOWN_POLLING_DELAY);
     } else {
+        m_node->context()->exit_status.store(EXIT_FAILURE);
```

Comment:
In commit "gui: return EXIT_FAILURE on post-init fatal errors" (624340f9acbdd3c2d54c3824248e2c0cff4cf487)

Not necessary to address here, but with #10102 when the node and GUI are running in separate processes, `m_node->context()` will be null, so this will crash.

You could fix and future-proof this by adding a `interfaces::Node::setExitStatus()` function, or just update the existing [`appInitMain()`](https://github.com/bitcoin/bitcoin/blob/153a6882f42fff3fdc63bf770d4c86a62c46c448/src/node/interfaces.cpp#L106-L109) function to set the error code:

```c++
    bool appInitMain(interfaces::BlockAndHeaderTipInfo* tip_info) override
    {
        if (!AppInitMain(*m_context, tip_info)) {
            m_context->exit_status.store(EXIT_FAILURE);
            return false;
        }
        return true;
    }
```

If you take let `appInitMain` set the error code, you could also simplify GUI code even more, skipping `Q_EMIT initializeResult(rv, tip_info)` when `appInitMain` fails and getting rid of the `rv` parameter.

Reply:
> You could fix and future-proof this by adding a interfaces::Node::setExitStatus() function, or just update the existing [appInitMain()](https://github.com/bitcoin/bitcoin/blob/153a6882f42fff3fdc63bf770d4c86a62c46c448/src/node/interfaces.cpp#L106-L109) function to set the error code.

Great, added it. Thanks!

> If you let appInitMain set the error code, you could also simplify GUI code even more, skipping Q_EMIT initializeResult(rv, tip_info) when appInitMain fails and getting rid of the rv parameter.

Hmm, that would be nice but, in a quick glance, I don't think that we can remove the `rv` parameter without introducing another signal to the `InitExecutor` class (which wouldn't be bad). Because we still need to call to the shutdown function at the GUI level: the `BitcoinApplication::requestShutdown` that is called inside `BitcoinApplication::initializeResult` when the rv parameter is false.

Still, even when I love to keep cleaning stuff, let's leave it for a follow-up so the gui related changes are minimal here.